### PR TITLE
5403: Mutating Operations Warnings

### PR DESCRIPTION
**Pull request by RosalineP**
Mutating Actions: CreateRunGroupAction, ToggleRunExperimentMembershipAction

### CODE REVIEW

File: query/src/org/labkey/query/QueryDefinitionImpl.java [hunk 1]
```diff
@@ -539,7 +540,10 @@ public TableInfo getTable(@NotNull UserSchema schema, @Nullable List<QueryExcept
             TableInfo table = _cache.get(key);
             if (table == null)
             {
-                table = createTable(schema, errors, includeMetadata, null, skipSuggestedColumns, allowDuplicateColumns);
```
Comment (labkey-adam) [inline query/src/org/labkey/query/QueryDefinitionImpl.java]: Add a comment with justification?

File: api/src/org/labkey/api/security/AuthenticationManager.java [hunk 2]
```diff
@@ -310,7 +311,13 @@ public ModelAndView getView(FORM form, BindException errors) throws Exception
             else
             {
                 HttpServletRequest request = getViewContext().getRequest();
-                PrimaryAuthenticationResult primaryResult = AuthenticationManager.finalizePrimaryAuthentication(request, response);
+
+                PrimaryAuthenticationResult primaryResult = null;
+                // Use instances provide their own authentication
```
[the remark this answers is not in the document]
Comment (RosalineP) [reply api/src/org/labkey/api/security/AuthenticationManager.java]: How about "SSO validation precludes need to monitor for mutating SQL vulnerabilities"?

File: experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java [hunk 3]
```diff
@@ -689,7 +702,7 @@ public NavTree appendNavTrail(NavTree root)
 //    @RequiresLogin
 //    public class SaveMaterialsAction extends MutatingApiAction<SaveMaterialsForm>
 //    {
-//        public ApiResponse execute(SaveMaterialsForm form, BindException errors) throws Exception
+//        public ApiResponse getView(SaveMaterialsForm form, BindException errors) throws Exception
```
[the remark this answers is not in the document]
Comment (RosalineP) [reply experiment/src/org/labkey/experiment/controllers/exp/ExperimentController.java]: Another instance of overzealous method name refactors. Must have forgotten to check over the comments on this after we chatted about that. Fixed!

File: api/src/org/labkey/api/security/AuthenticationManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (labkey-adam) [reply api/src/org/labkey/api/security/AuthenticationManager.java]: Maybe "Some SSO protocols allow GET, but validation requires a secret so it's not susceptible to CSRF attacks"

[the remark this answers is not in the document]
Comment (RosalineP) [reply api/src/org/labkey/api/security/AuthenticationManager.java]: Sounds good!

File: query/src/org/labkey/query/QueryDefinitionImpl.java (hunk 1 above)
Comment (RosalineP) [reply query/src/org/labkey/query/QueryDefinitionImpl.java]: Done!

**Pull request merged**